### ISSUE TITLE
L-BFGS for style transfer

### ISSUE THREAD

**Issue by nsthorat** [tensorflow/tfjs#127]
_From @carlthome on February 15, 2018 23:11_

[Fast style transfer](https://github.com/reiinakano/fast-style-transfer-deeplearnjs) with pretrained models is __really__ cool, but the original optimization-based method presented by Gatys et al. seems like such a great fit for what deeplearn.js could provide with its WebGL integration and builtin gradient computation.

I'd really like to make interactive style transfer web apps but there's no L-BFGS optimizer in deeplearn.js. Is that a planned addition or will you focus on SGD variants?

_Copied from original issue: tensorflow/tfjs-core#734_

**Comment by nsthorat** [comment]
Currently we're only focusing on SGD variants of optimizers, but if you are interested in making an L-BFGS, 2nd order gradient based optimizer you can either
1) extend the Optimizer base class and override `computeGradients`, and `applyGradients`
or
2) simply write your own optimizer without using the Optimizer class. You can compute gradients with the eager style method of differentiation (use methods from [this section](http://deeplearnjs.org/docs/api/#Training-Gradients) and then use `dl.variable()` and `variable.assign()` to do an update.

If you decide you're up for it, let us know if you run into any issues!

**Comment by rthadur** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
